### PR TITLE
[WIP] EDXOLDMNG update old mogo tests

### PR DESCRIPTION
**Pull request by UvgenGen**
- opportunistic removed of ModuleStoreEnum.Type.mongo in cases where both it and ModuleStoreEnum.Type.split are being tested in the same test (e.g. via ddt)

### CODE REVIEW

File: xmodule/modulestore/mongo/draft.py [hunk 1]
```diff
@@ -143,25 +133,7 @@ def has_item(self, usage_key, revision=None):  # lint-amnesty, pylint: disable=a
...
-                self.get_branch_setting() == ModuleStoreEnum.Branch.published_only
-        ):
-            return has_published()
-        elif revision is None:
-            key = usage_key.to_deprecated_son(prefix='_id.')
-            del key['_id.revision']
-            return self.collection.count_documents(key) > 0
-        else:
-            raise UnsupportedRevisionError()
+        return False
```
Comment (GlugovGrGlib) [inline xmodule/modulestore/mongo/draft.py:136]: To check if we need to add backward compatibility for this method

File: xmodule/modulestore/mongo/base.py [hunk 2]
```diff
@@ -1044,7 +1044,7 @@ def get_courses(self, **kwargs):
             ],
             []
         )
-        return [course for course in base_list if not isinstance(course, ErrorBlock)]
+        return []
```
Comment (cmltaWt0) [inline xmodule/modulestore/mongo/base.py:1047]: Do we need a comment here?

File: cms/djangoapps/contentstore/views/course.py [hunk 3]
```diff
@@ -275,6 +275,9 @@ def course_handler(request, course_key_string=None):
         if response_format == 'json' or 'application/json' in request.META.get('HTTP_ACCEPT', 'application/json'):
             if request.method == 'GET':
                 course_key = CourseKey.from_string(course_key_string)
+                # raise 404 if old mogo course
```
Comment (cmltaWt0) [inline cms/djangoapps/contentstore/views/course.py:278]: mogo -> mongo

Let's change it to `# raise 404 for old mongo course`

File: cms/djangoapps/contentstore/views/course.py [hunk 4]
```diff
@@ -292,7 +295,11 @@ def course_handler(request, course_key_string=None):
             if course_key_string is None:
                 return redirect(reverse('home'))
             else:
-                return course_index(request, CourseKey.from_string(course_key_string))
+                course_key = CourseKey.from_string(course_key_string)
+                # raise 404 if old mogo course
```
Comment (cmltaWt0) [inline cms/djangoapps/contentstore/views/course.py:299]: The same as above

File: xmodule/modulestore/mixed.py [hunk 5]
```diff
@@ -295,6 +294,11 @@ def get_course_summaries(self, **kwargs):
                     log.warning(
                         "Modulestore %s have duplicate courses %s; skipping from result.", store, course_id
                     )
+                # Skip course if it's from old mongo modulestore.
+                elif course_id.deprecated:
+                    log.warning(
+                        "Course %s from %s; skipping from result.", course_id, store
```
Comment (cmltaWt0) [inline xmodule/modulestore/mixed.py:300]: Let's add more descriptive log string to mention that course is skipped due to it's oldesty :)

File: xmodule/modulestore/mixed.py [hunk 6]
```diff
@@ -717,17 +721,9 @@ def clone_course(self, source_course_id, dest_course_id, user_id, fields=None, *
...
-            split_migrator.migrate_mongo_course(source_course_id, user_id, dest_course_id.org,
-                                                dest_course_id.course, dest_course_id.run, fields, **kwargs)
-
-            # the super handles assets and any other necessities
-            super().clone_course(source_course_id, dest_course_id, user_id, fields, **kwargs)
-        else:
-            raise NotImplementedError("No code for cloning from {} to {}".format(
-                source_modulestore, dest_modulestore
-            ))
+        raise NotImplementedError("No code for cloning from {} to {}".format(
```
Comment (cmltaWt0) [inline xmodule/modulestore/mixed.py:724]: Could we use f-string here?

**Comment by cmltaWt0** [comment]
In general - there is a lot of non tests changes. Probabkly we need to change the title.

**Comment by UvgenGen** [comment]
@cmltaWt0 I rebased this branch, now only the necessary changes are here.

**Pull request closed without merging**